app/lib/actions/pulp3/repository/upload_file.rb

```diff
@@ -0,0 +1,84 @@
+module Actions
+  module Pulp3
+    module Repository
+      class UploadFile < Pulp3::AbstractAsyncTask
+        def plan(repository, smart_proxy, file)
+          plan_self(:repository_id => repository.id, :smart_proxy_id => smart_proxy.id, :file => file)
+        end
+
+        def invoke_external_task
+          repo = ::Katello::Repository.find(input[:repository_id])
+          repo_backend_service = repo.backend_service(smart_proxy)
+          upload_class = repo_backend_service.upload_class
+          uploads_api = repo_backend_service.uploads_api
+          offset = 0
+          response = ""
+          File.open(input[:file], "rb") do |file|
+            total_size = File.size(file)
+            upload_data = upload_class.new({size: total_size})
+            response = uploads_api.create(upload_data)
+            upload_href = response._href
+            chunksize = upload_chunk_size
+            offset = 0
+            sha256 = Digest::SHA256.new
```

Comment:
would it be simpler to just do something like:
```
Digest::SHA256.file(filename)
```

instead of doing this append method?